packages/server/lib/wallet.js

```diff
@@ -126,16 +126,17 @@ function sendCoinsBatch(settings, txs, cryptoCode) {
     })
 }
 
-function newAddress(settings, info, tx) {
-  const walletAddressPromise = fetchWallet(settings, info.cryptoCode).then(r =>
-    r.wallet.newAddress(r.account, info, tx),
-  )
+function newAddress(settings, info) {
+  const walletAddressPromise = fetchWallet(settings, info.cryptoCode)
+    .then(r => r.wallet.newAddress(r.account, info))
+    .then(res => (typeof res === 'string' ? { address: res } : res))
```

Comment:
if we want to return an object, every wallet should return an object. Ideally return the wallet ID for the wallets we can generate the ID from.